### PR TITLE
test_linearizer_failures: add failing kernel from GPT2 CUDA

### PR DESCRIPTION
**Pull request by flammit**
this bisects to #3530 which I think caused the uops to be incorrect on all platforms

### CODE REVIEW

File: test/test_linearizer_failures.py [hunk 1]
```diff
@@ -186,5 +186,11 @@ def test_failure_24(self):
...
     helper_test_lin(Linearizer(ast), opts, failed_platforms=[])
 
+  # this is the cause of the GPT2 BEAM instability. bisects to PR#3530 O(n) arange attempt
+  def test_failure_25(self):
+    ast = LazyOp(op=BufferOps.STORE, src=(LazyOp(op=BinaryOps.ADD, src=(LazyOp(op=ReduceOps.SUM, src=(LazyOp(op=BufferOps.CONST, src=(), arg=ConstBuffer(val=1, dtype=dtypes.int, st=ShapeTracker(views=(View(shape=(1025, 2047), strides=(0, 0), offset=0, mask=((0, 1025), (1023, 2047)), contiguous=False), View(shape=(1024, 1024), strides=(1, 2048), offset=0, mask=None, contiguous=False))))),), arg=(1,)), LazyOp(op=BufferOps.CONST, src=(), arg=ConstBuffer(val=-1, dtype=dtypes.int, st=ShapeTracker(views=(View(shape=(1024, 1), strides=(0, 0), offset=0, mask=None, contiguous=False),))))), arg=None),), arg=MemBuffer(idx=0, dtype=dtypes.int, st=ShapeTracker(views=(View(shape=(1024, 1), strides=(1, 0), offset=0, mask=None, contiguous=True),))))
+    opts = [Opt(op=OptOps.GROUP, axis=0, amt=16), Opt(op=OptOps.UNROLL, axis=0, amt=4)]
+    helper_test_lin(Linearizer(ast), opts, failed_platforms=["METAL", "GPU", "HIP", "HSA", "CUDA"])
```
Comment (chenyuxyz) [inline test/test_linearizer_failures.py:193]: nice find!

btw "HIP" was removed. The one in CI is "RHIP" now. Can you update this and the "HIP" above too

Comment (flammit) [reply test/test_linearizer_failures.py:193]: Yup will do!

**Comment by flammit** [comment]
@patosai could you take a look at this regression and see what fix can be implemented?

**Comment by patosai** [comment]
pikachu

**Pull request merged**